### PR TITLE
stats: Document OpenMetrics

### PR DESCRIPTION
**Pull request by cmouse**
JIRA: DOV-3931

Documentation for the new openmetrics exporter.

### CODE REVIEW

File: source/configuration_manual/stats/openmetrics.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+===================================
+OpenMetrics exporter for statistics
+===================================
+
+.. versionadded:: 2.3.11
+
+Basic configuration
+===================
+
+Dovecot has support for `OpenMetrics <https://openmetrics.io/>`_ exposition for statistics.
```
Comment (sirainen) [inline source/configuration_manual/stats/openmetrics.rst:10]: "exposition" word looked weird, but googling it I see it's a thing. But usually it's together as "exposition format", so maybe add format?

File: source/configuration_manual/stats/openmetrics.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+This can be enabled by adding following configuration::
```
Comment (sirainen) [inline source/configuration_manual/stats/openmetrics.rst:11]: is two :: intentional?..

File: source/configuration_manual/stats/openmetrics.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+       port = 9900
+     }
+   }
+
+This will enable Dovecot to expose all configured metrics in OpenMetrics format on ``http://host:9900/metrics`` using `text-based format <https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format>`_ .
+
+Statistics format
+=================
+
+By default, Dovecot exposes any configured metrics.
```
Comment (sirainen) [inline source/configuration_manual/stats/openmetrics.rst:24]: s/any/all/?

File: source/configuration_manual/stats/openmetrics.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+The name of the base metric is used as above, and any dynamically generated sub-metrics are exported using labels.
+Quantized sub-metrics are exported as histograms.
+
+Dovecot will also export version and uptime as special metrics even if nothing is configured.
+These are called ``dovecot_stats_uptime_seconds`` and ``dovecot_build_info``.
+
+Example page
+============
+
+::
```
Comment (sirainen) [inline source/configuration_manual/stats/openmetrics.rst:39]: `code-block::`?

File: source/configuration_manual/stats/openmetrics.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+Dynamically generated statistics with :ref:`group_by <statistics_group_by>` will be exported too.
+The name of the base metric is used as above, and any dynamically generated sub-metrics are exported using labels.
+Quantized sub-metrics are exported as histograms.
+
+Dovecot will also export version and uptime as special metrics even if nothing is configured.
+These are called ``dovecot_stats_uptime_seconds`` and ``dovecot_build_info``.
+
+Example page
+============
```
Comment (sirainen) [inline source/configuration_manual/stats/openmetrics.rst:37]: could have an example about named groups, like imap_command_finished with multiple names

File: source/configuration_manual/stats/openmetrics.rst (hunk 2 above)
Comment (cmouse) [reply source/configuration_manual/stats/openmetrics.rst:11]: yes

Comment (cmouse) [reply source/configuration_manual/stats/openmetrics.rst:11]: shorthand for code-block

Comment (sirainen) [reply source/configuration_manual/stats/openmetrics.rst:11]: But it's not used anywhere in the repository now. Should stay consistent.

Comment (sirainen) [reply source/configuration_manual/stats/openmetrics.rst:11]: oh, uh, what, I thought I was answering about another thing

Comment (sirainen) [reply source/configuration_manual/stats/openmetrics.rst:11]: or I guess it's basically the same

Comment (sirainen) [reply source/configuration_manual/stats/openmetrics.rst:11]: But I see it's used in some other places already. So .. I suppose it's okay then. Just will be difficult to grep for them if we need to do things like the idea of converting code-blocks to that other type that allows links.

**Pull request merged**